independent-projects/bootstrap/app-model/src/main/java/io/quarkus/bootstrap/util/IoUtils.java

```diff
@@ -89,6 +90,36 @@ public FileVisitResult postVisitDirectory(Path dir, IOException e)
         }
     }
 
+    public static void recursiveDeleteAndThenCreate(Path root) throws IOException {
```

Comment:
Isn't easier to name it `recreate`?

Reply:
I'm horrible at naming, so whatever you guys think

Reply:
Given that there could be content, re-create could also be ambiguous. If we are looking for a shorter name it could be `cleanDir`. And the implementation instead of actually removing the root dir, simply remove its content instead.

Reply:
I tend to agree that this shouldn't be named `recreate`. So I'll merge for now and we can always improve thw name